crates/polars-time/src/truncate.rs

```diff
@@ -31,31 +23,61 @@ pub trait PolarsTruncate {
-        let every = Duration::parse(&options.every);
-        let offset = Duration::parse(&options.offset);
-        let w = Window::new(every, every, offset);
+        let offset = Duration::parse(offset);
 
         let func = match self.time_unit() {
             TimeUnit::Nanoseconds => Window::truncate_ns,
             TimeUnit::Microseconds => Window::truncate_us,
             TimeUnit::Milliseconds => Window::truncate_ms,
         };
 
-        let out = match ambiguous.len() {
-            1 => match ambiguous.get(0) {
-                Some(ambiguous) => self
-                    .0
-                    .try_apply(|timestamp| func(&w, timestamp, tz, ambiguous)),
-                _ => Ok(self.0.apply(|_| None)),
+        let out = match (every.len(), ambiguous.len()) {
+            (1, 1) => match (every.get(0), ambiguous.get(0)) {
+                (Some(every), Some(ambiguous)) => {
+                    let every = Duration::parse(every);
+                    let w = Window::new(every, every, offset);
+                    self.0
+                        .try_apply(|timestamp| func(&w, timestamp, tz, ambiguous))
+                },
+                _ => Ok(Int64Chunked::full_null(self.name(), self.len())),
+            },
+            (1, _) => {
+                if let Some(every) = every.get(0) {
+                    let every = Duration::parse(every);
+                    let w = Window::new(every, every, offset);
+                    try_binary_elementwise_values(
+                        self,
+                        ambiguous,
+                        |timestamp: i64, ambiguous: &str| func(&w, timestamp, tz, ambiguous),
+                    )
+                } else {
+                    Ok(Int64Chunked::full_null(self.name(), self.len()))
+                }
+            },
+            (_, 1) => {
+                if let Some(ambiguous) = ambiguous.get(0) {
+                    try_binary_elementwise(self, every, |opt_timestamp, opt_every| {
+                        match (opt_timestamp, opt_every) {
+                            (Some(timestamp), Some(every)) => {
+                                let every = Duration::parse(every);
+                                let w = Window::new(every, every, offset);
+                                func(&w, timestamp, tz, ambiguous).map(Some)
+                            },
+                            _ => Ok(None),
+                        }
+                    })
+                } else {
+                    Ok(Int64Chunked::full_null(self.name(), self.len()))
+                }
             },
             _ => {
-                try_binary_elementwise_values(self, ambiguous, |timestamp: i64, ambiguous: &str| {
-                    func(&w, timestamp, tz, ambiguous)
-                })
+                polars_bail!(
+            ComputeError: "At least one of ambiguous and every requires a length of 1. Length of ambiguous `{}`, every `{}`", ambiguous.len(), every.len())
```

Comment:
is there any theoretical reason why both `every` and `ambiguous` can't be a full column, or is it just that there currently isn't a `try_ternary_elementwise`?

if the latter, then perhaps this could be a `NotImplementedError`?